aiotfm/client.py

```diff
@@ -599,13 +608,46 @@ def dispatch(self, event, *args, **kwargs):
 		while not self.main.socket.connected:
 			await asyncio.sleep(.1)
 
-		packet = Packet.new(28, 1).write16(keys.version).writeString(keys.connection)
+	async def sendHandshake(self):
+		"""|coro|
+		Sends the handshake packet so the server recognizes this socket as a player.
+		"""
+		packet = Packet.new(28, 1).write16(self.keys.version).writeString(self.keys.connection)
 		packet.writeString('Desktop').writeString('-').write32(0x1fbd).writeString('')
 		packet.writeString('74696720697320676f6e6e61206b696c6c206d7920626f742e20736f20736164')
 		packet.writeString("A=t&SA=t&SV=t&EV=t&MP3=t&AE=t&VE=t&ACC=t&PR=t&SP=f&SB=f&DEB=f&V=LNX 29,0,0,140&M=Adobe Linux&R=1920x1080&COL=color&AR=1.0&OS=Linux&ARCH=x86&L=en&IME=t&PR32=t&PR64=t&LS=en-US&PT=Desktop&AVD=f&LFD=f&WD=f&TLS=t&ML=5.1&DP=72")
 		packet.write32(0).write32(0x6257).writeString('')
 
 		await self.main.send(packet)
+
+	async def start(self, api_tfmid, api_token, keys=None):
+		"""|coro|
+		Starts the client.
+
+		:param api_tfmid: :class:`int` or :class:`str` your Transformice id.
+		:param api_token: :class:`str` your token to access the API.
+		"""
+		if keys is not None:
+			self.keys = keys
+		else:
+			self.api_tfmid = api_tfmid

```

Comment:
I don't like storing the token as a Client attribute. But I don't see how it could be done otherwise.

Reply:
The same thing happened with the user and password. I hate storing them here and that's why I disabled the run method + auto restart.